src/main/java/org/opensearch/agent/tools/PPLTool.java

```diff
@@ -83,12 +87,46 @@ public class PPLTool implements Tool {
 
     private String contextPrompt;
 
+    private PPLModelType pplModelType;
+
     private static Gson gson = new Gson();
 
-    public PPLTool(Client client, String modelId, String contextPrompt) {
+    private static Map<String, String> defaultPromptDict;
+
+    static {
+        try {
+            defaultPromptDict = loadDefaultPromptDict();
+        } catch (IOException e) {
+            throw new RuntimeException("fail to load default prompt dict" + e.getMessage());
+        }
+    }
+
+    private enum PPLModelType {
+        CLAUDE,
+        FINETUNE;
+
+        public static PPLModelType from(String value) {
+            if (value.isEmpty()) {
+                return PPLModelType.CLAUDE;
+            }
+            try {
+                return PPLModelType.valueOf(value.toUpperCase(Locale.ROOT));
+            } catch (Exception e) {
+                throw new IllegalArgumentException("Wrong PPL Model type, should be CLAUDE or FINETUNE");
+            }
+        }
+
+    }
+
+    public PPLTool(Client client, String modelId, String contextPrompt, String pplModelType) {
         this.client = client;
         this.modelId = modelId;
-        this.contextPrompt = contextPrompt;
+        this.pplModelType = PPLModelType.from(pplModelType);
+        if (contextPrompt.isEmpty()) {
+            this.contextPrompt = this.defaultPromptDict.getOrDefault(this.pplModelType.toString(), "");
```

Comment:
Is it possible this pplModelType.toString() produce NPE?

Reply:
Since we init the modeltype with the default value, the model type should be an object or throw an error. So it shouldn't be a NPE.

Reply:
Ok, but there's no need to invoke toString() on a String class, let's remove this unnecessary call.